src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,181 +1,173 @@
-import React, { useCallback, useEffect, useState } from "react";
+import React, { useCallback, useState, ChangeEvent, useMemo } from "react";
+import classNames from "classnames";
+import { debounce } from "lodash";
+
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+
+import { DEFAULT_PARTY_NAME } from "settings";
 
 import { VenueEvent } from "types/venues";
 import { Room, RoomTypes } from "types/rooms";
-import { User } from "types/User";
 
-import { WithId } from "utils/id";
-import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 import { isTruthy } from "utils/types";
+import { uppercaseFirstChar } from "utils/string";
+import { formatUtcSecondsRelativeToToday } from "utils/time";
+import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 
 import { useWorldUsers } from "hooks/users";
 import { useSelector } from "hooks/useSelector";
 import { useProfileModalControls } from "hooks/useProfileModalControls";
 
 import { RoomModal } from "components/templates/PartyMap/components";
 
-import "./NavSearchBar.scss";
 import { InputField } from "components/atoms/InputField";
-import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
-interface SearchResult {
-  rooms: Room[];
-  users: readonly WithId<User>[];
-  events: VenueEvent[];
-}
+import CloseIcon from "assets/icons/nav-dropdown-close.png";
+
+import { NavSearchResult } from "./NavSearchResult";
+
+import "./NavSearchBar.scss";
+
+const emptyEventsArray: VenueEvent[] = [];
+
+const DEBOUNCE_TIME = 200;
 
 const NavSearchBar = () => {
   const [searchQuery, setSearchQuery] = useState("");
-  const onSearchInputChange = useCallback(
-    (e) => setSearchQuery(e.target.value),
+
+  const debouncedSearch = useMemo(
+    () =>
+      debounce(
+        (str: string) => setSearchQuery(str.toLowerCase()),
+        DEBOUNCE_TIME
+      ),
     []
   );
```

Comment:
Do we only use this function from the input field? If so, instead of passing through the string, we could probably pass through the event itself. That way we can fully memoize it here, and not need an inline function handler later on (which will cause re-renders)

Reply:
I left 2 functions but applied useMemo and useCallback. If we try to use `debounce` doesn't work with `ChangeEvent` correctly, because by the time dobounced function is executing there is no event object.

```ts
  const debouncedSearch = useMemo(
    () =>
      debounce((value: string) => {
        setSearchQuery(value.toLowerCase());
      }, DEBOUNCE_TIME),
    []
  );

  const onSearchInputChange = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      const value = e.target.value;
      debouncedSearch(value);
    },
    [debouncedSearch]
  );

```

Reply:
see the commit: https://github.com/sparkletown/sparkle/pull/1328/commits/c7104e118d01c1d1d5100c9cf01154be7115c42e

Reply:
Makes sense. Sounds good.